### ISSUE TITLE
Integrate code from regtricks?

### ISSUE THREAD

**Issue by tomfrankkirk** [poldracklab/nitransforms#100]
Hi, I've been pointed here by a user not on GitHub. I've been working on a toolbox that overlaps with your aims on this project, especially with regards to interfacing with FSL. This is it: https://github.com/tomfrankkirk/regtricks

The aim is to allow for easy combining and application of registration transforms, without replacing the process of finding registrations themselves. As of yet, it interfaces with the main FSL tools: FLIRT, MCFLIRT, FNIRT 

Internally, it converts all registrations into world-world convention (as is the case for BIDS X5, for which there is some support built in) and then uses scipy to perform all interpolation / resampling operations. You can chain together an arbitrary number of transformations and it will resolve them into a single overall transformation object that is applied with one interpolation step. The user has full control over this if they want it. 

So, in summary, its an elaborate wrapper for reading in FSL transforms and passing them off to scipy's interpolation libraries. If people were willing to help add code to interface with other toolboxes (ANTS, SPM etc) then I think all the underlying logic for combining / applying transforms would still hold true, and hopefully thats a step in the right direction for your overall aims. Equally you're very welcome to pull code out of this if helpful.  

Let me know if you'd like to talk more. 
tom

**Comment by tomfrankkirk** [comment]
@oesteban Hello there - not sure if you've seen the above, and not sure if this was the best way to approach you. Hope you find this interesting.

**Comment by oesteban** [comment]
@tomfrankkirk Hi Tom, happy to chat more. This project though is envisioned to be merged into nibabel and to be general (not just FSL).

Most of the features you mention are already implemented here, or will be shortly. Please let me know how do you think both projects can be coordinated.

**Comment by tomfrankkirk** [comment]
@oesteban  Issue #89: extend the collapse method for a transform chain to include non-linear. 

I took a different approach to chaining transforms: although the concept exists in my code, I don't actually implement it as a specific class but instead use recursion and type promotion. 

The general idea is: 
1) a function chain() that operates on a list of transforms. for just two transforms, it performs the multiplication B @ A explicitly, otherwise it calls itself recursively with a subset of the list until it reaches the base case.  https://github.com/tomfrankkirk/regtricks/blob/daf02bca67bb11e89ed0ff20973960a8e1643b2a/regtricks/multiplication.py#L202

2) a set of rules for multiplying transformations that lead to type promotion. for any two transforms, the result will be of the same type as the 'highest' of the inputs, where the hierarchy is (low -> high): linear registration, motion correction, non-linear reg, non-linear motion correction. This means any multiplication of two transforms always leads to a new transform. https://github.com/tomfrankkirk/regtricks/blob/daf02bca67bb11e89ed0ff20973960a8e1643b2a/regtricks/multiplication.py#L12

using 1) and 2), any chain of transformations is evaluated immediately and returned as a new transform object. In effect, the collapse() call is performed straight away and the 'chain of transforms' exists only in the calling scope (as the actual list that the user provides to chain()). 

the only caveat to the above is that my implementation is perhaps FSL-specific for the non-linear components (in particular, the concept of a pre-mat and post-mat). I don't know if other non-linear tools make use of the same concepts and therefore if it would port over nicely.
